tools/tune/eq/gui.m

```diff
@@ -1,7 +1,37 @@
 global EQ = [];
+global main_panes;
+fn = 1;
 
-function update_plot(channel)
+function peq = build_peq(channel)
+  global EQ;
+  peq = [];
+  for i = 1:8
+    biquad = EQ(i, channel);
+    if get(biquad.enable_switch, "value") == 1
+      type = get(biquad.type.dropdown, "value");
+      freq = str2num(get(biquad.freq.edit, "string"));
+      Q = str2double(get(biquad.Q.edit, "string"));
+      gain = str2double(get(biquad.gain.edit, "string"));
+      % dropdown maps straight to filter index
+      % filter only uses needed arguments, all others will be ignored
+      peq = [peq; type, freq, gain, Q];
+    endif
+  endfor
+end
 
+function update_plot(channel)
+  global main_panes;
+  pl = main_panes(channel).plot;
+  Fs = 44100;
```

Comment:
The sample rate is hard coded here. Could it be set via GUI or retrieved from some config file?